.github/workflows/rust.yml

```diff
@@ -0,0 +1,46 @@
+name: Rust
+
+on:
+  push:
+    branches: [ imlk-dev ]
+  pull_request:
+    branches: [ imlk-dev ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Setup components
+      run: rustup component add clippy rustfmt
+    - name: Check
+      shell: bash
+      run: |
```

Comment:
Why not run the githook instead of duplicating the script here?

Reply:
The two scripts are slightly different. The bash script in `pre-commit` will only check staged files and format the files, while the script in `rust.yml` will check the files changed between `github.event.before` and `github.event.after` but does not format.

Or should I merge the two into one script file and put it in `scripts/check-format.sh`?

Reply:
They can stay separate, but I like the idea of having a standalone script that can be run locally.

Reply:
> They can stay separate, but I like the idea of having a standalone script that can be run locally.

In order to run locally, how about running `cargo fmt` directly?

Reply:
That works for me. But how would that be different from the pre-commit hook?

Reply:
> That works for me. But how would that be different from the pre-commit hook?

When submitting, the pre-commit hook script will automatically format the staged code and remind you in a friendly way. 
here is an example:

[![asciicast](https://asciinema.org/a/iD8QrzTpTuoinFsq4Q462l9zx.svg)](https://asciinema.org/a/iD8QrzTpTuoinFsq4Q462l9zx)

Reply:
That looks good to me.

Reply:
> That looks good to me. Is this ready to merge?

Yes